keras_cv/models/backbones/csp_darknet/csp_darknet_backbone.py

```diff
@@ -0,0 +1,385 @@
+            {
+                "include_rescaling": include_rescaling,
+                "input_shape": input_shape,
+                "input_tensor": input_tensor,
+            }
+        )
+        return CSPDarkNetBackbone.from_preset("cspdarknets", **kwargs)
+
+    @classproperty
+    def presets(cls):
+        """Dictionary of preset names and configurations."""
+        return {}
+
+
+class CSPDarkNetMBackbone(CSPDarkNetBackbone):
+    def __new__(
+        cls,
+        include_rescaling=True,
+        input_shape=(None, None, 3),
+        input_tensor=None,
+        **kwargs,
+    ):
+        # Pack args in kwargs
+        kwargs.update(
+            {
+                "include_rescaling": include_rescaling,
+                "input_shape": input_shape,
+                "input_tensor": input_tensor,
+            }
+        )
+        return CSPDarkNetBackbone.from_preset("cspdarknetm", **kwargs)
+
+    @classproperty
+    def presets(cls):
+        """Dictionary of preset names and configurations."""
+        return {}
+
+
+class CSPDarkNetLBackbone(CSPDarkNetBackbone):
+    def __new__(
+        cls,
+        include_rescaling=True,
+        input_shape=(None, None, 3),
+        input_tensor=None,
+        **kwargs,
+    ):
+        # Pack args in kwargs
+        kwargs.update(
+            {
+                "include_rescaling": include_rescaling,
+                "input_shape": input_shape,
+                "input_tensor": input_tensor,
+            }
+        )
+        return CSPDarkNetBackbone.from_preset("cspdarknetl", **kwargs)
+
+    @classproperty
+    def presets(cls):
+        """Dictionary of preset names and configurations."""
+        return {
+            "cspdarknetl_imagenet": copy.deepcopy(
+                backbone_presets["cspdarknetl_imagenet"]
+            )
+        }
+
+    @classproperty
+    def presets_with_weights(cls):
+        """Dictionary of preset names and configurations that include
+        weights."""
+        return cls.presets
+
+
+class CSPDarkNetXBackbone(CSPDarkNetBackbone):
+    def __new__(
+        cls,
+        include_rescaling=True,
+        input_shape=(None, None, 3),
+        input_tensor=None,
+        **kwargs,
+    ):
+        # Pack args in kwargs
+        kwargs.update(
+            {
+                "include_rescaling": include_rescaling,
+                "input_shape": input_shape,
+                "input_tensor": input_tensor,
+            }
+        )
+        return CSPDarkNetBackbone.from_preset("cspdarknetl", **kwargs)
+
+    @classproperty
+    def presets(cls):
+        """Dictionary of preset names and configurations."""
+        return {}
+
+
+setattr(
+    CSPDarkNetTinyBackbone,
+    "__doc__",
+    ALIAS_DOCSTRING.format(
+        name="Tiny", channels="[48, 96, 192, 384]", depths="[1, 3, 3, 1]"
+    ),
+)
+setattr(
+    CSPDarkNetSBackbone,
+    "__doc__",
+    ALIAS_DOCSTRING.format(
+        name="S", channels="[64, 128, 256, 512]", depths="[1, 3, 3, 1]"
```

Comment:
@LukeWood @quantumalaviya are these depth parameters the expected values for YOLOx?

They are different than the ones used for YOLOv8. (If it turns out the two use different configs, we can just create separate presets)